include/boost/compute/context.hpp

```diff
@@ -175,27 +175,13 @@ class context
     /// devices, the first is returned.
     device get_device() const
     {
-        size_t count = 0;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         0,
-                         0,
-                         &count);
-        if(count == 0){
-            return device();
-        }
+        std::vector<device> devices = get_devices();
 
-        cl_device_id id;
-        clGetContextInfo(m_context,
-                         CL_CONTEXT_DEVICES,
-                         sizeof(cl_device_id),
-                         &id,
-                         0);
-        if(id == 0){
+        if(devices.empty()) {
             return device();
         }
 
-        return device(id);
+        return devices.front();
```

Comment:
Hmm, was there actually a bug in this code? What errors were you seeing?

The code as it is now only returns the first device. Changing it to call `get_devices()` will perform a dynamic-memory allocation (to create a `std::vector<device>`) which is unnecessary if we only need the first device for the context.